packages/typedoc-plugin-appium/lib/converter/converter.ts

```diff
@@ -0,0 +1,315 @@
+        if (cmdInfo.hasCommands) {
+          projectCommands.set(mod, this.#convertModuleClasses(mod));
+        }
+        this.#log.info('Converted module %s', mod.name);
+      }
+    } else {
+      projectCommands.set(project, this.#convertModuleClasses(project));
+    }
+
+    this.#log.info('Found commands in %d module(s)', projectCommands.size);
+
+    return projectCommands;
+  }
+
+  #convertBaseDriver(baseDriver: BaseDriverDeclarationReflection): CommandInfo {
+    const baseDriverRoutes = this.#convertMethodMap(baseDriver);
+    if (!baseDriverRoutes.size) {
+      throw new TypeError(`Could not find any commands in BaseDriver!?`);
+    }
+
+    // no execute commands in BaseDriver
+    return new CommandInfo(baseDriverRoutes);
+  }
+
+  #convertExecuteMethodMap(refl: DeclarationReflection): ExecuteCommandSet {
+    const executeMethodMap = refl.getChildByName(NAME_EXECUTE_METHOD_MAP);
+    const commandRefs: ExecuteCommandSet = new Set();
+    if (
+      executeMethodMap?.flags.isStatic &&
+      isDeclarationReflection(executeMethodMap) &&
+      isReflectionType(executeMethodMap.type)
+    ) {
+      for (const newMethodProp of executeMethodMap.type.declaration.getChildrenByKind(
+        ReflectionKind.Property
+      )) {
+        const comment = newMethodProp.comment;
+        const script = newMethodProp.originalName;
+        if (isDeclarationReflection(newMethodProp) && isReflectionType(newMethodProp.type)) {
+          const commandProp = newMethodProp.type.declaration.getChildByName(NAME_COMMAND);
+          if (isDeclarationReflection(commandProp) && isLiteralType(commandProp.type)) {
+            const command = String(commandProp.type.value);
+            const paramsProp = newMethodProp.type.declaration.getChildByName(NAME_PARAMS);
+            let requiredParams: string[] = [];
+            let optionalParams: string[] = [];
+            if (isDeclarationReflection(paramsProp)) {
+              requiredParams = this.#parseRequiredParams(paramsProp);
+              optionalParams = this.#parseOptionalParams(paramsProp);
+            }
+            commandRefs.add({
+              command,
+              requiredParams,
+              optionalParams,
+              script,
+              comment,
+            });
+          }
+        }
+      }
+    }
+    return commandRefs;
+  }
+
+  /**
+   * Extracts information about `MethodMap` objects
+   * @param refl - Some reflection we want to inspect. Could refer to a module or a class
+   * @returns Lookup of routes to {@linkcode CommandMap} objects
+   */
+  #convertMethodMap(refl: DeclarationReflection): RouteMap {
+    const routes: RouteMap = new Map();
+
+    let methodMap: MethodMapDeclarationReflection;
+    const child = isBaseDriverDeclarationReflection(refl)
+      ? refl.getChildByName(NAME_METHOD_MAP)
+      : refl.getChildByName(NAME_NEW_METHOD_MAP);
+
+    if (isMethodMapDeclarationReflection(child)) {
```

Comment:
deeply nested conditions are hard to parse. Consider changing them to guard conditions if possible, for example:

```
if (x) {a1()} else {return a2()}
```

to 

```
if (!x) {
  return a2()
};

a1()
```